pkg/auth/server/login/login.go

```diff
@@ -133,19 +159,98 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 	l.auth.AuthenticationSucceeded(context, then, w, req)
 }
 
-var DefaultLoginFormRenderer = loginTemplateRenderer{}
+// NewLoginFormRenderer creates a login form renderer that takes in an optional custom template to
+// allow branding of the login page. Uses the default if customLoginTemplateFile is not set.
+func NewLoginFormRenderer(templates *configapi.OAuthTemplates) (*loginTemplateRenderer, error) {
+	r := &loginTemplateRenderer{}
+	if templates != nil && len(templates.Login) > 0 {
+		customTemplate, err := template.ParseFiles(templates.Login)
+		if err != nil {
+			return nil, err
+		}
+		r.loginTemplate = customTemplate
+	} else {
+		r.loginTemplate = defaultLoginTemplate
+	}
+
+	return r, nil
+}
+
+func ValidateLoginTemplate(templates *configapi.OAuthTemplates) fielderrors.ValidationErrorList {
```

Comment:
this should take the template content ([]byte) only... and we should probably have a unit test that tests the default template content using it as well